File: supplemental/pocketbase/pb_migrations/initial_settings.js
/// <reference path="../pb_data/types.d.ts" />
migrate(
  (app) => {
    // initial settings
    const settings = app.settings();

    settings.meta.appName = "Cially";
    settings.meta.hideControls = true;
    settings.logs.minLevel = 4;

    app.save(settings);

    // create superuser
    const superusers = app.findCollectionByNameOrId("_superusers");

    const record = new Record(superusers);

    record.set(
      "email",
      $os.getenv("POCKETBASE_ADMIN_EMAIL") || "admin@cially.org",
    );
    record.set("password", $os.getenv("POCKETBASE_ADMIN_PASSWORD") || "admin123!");

    app.save(record);
  },
  (app) => {
    // add down queries...
  },
);

File: apps/pocketbase/migrations/initial_settings.go
package migrations

import (
	"os"
	"github.com/pocketbase/pocketbase/core"
	m "github.com/pocketbase/pocketbase/migrations"
)

var (
	adminEmail    = os.Getenv("POCKETBASE_ADMIN_EMAIL")
	adminPassword = os.Getenv("POCKETBASE_ADMIN_PASSWORD")
)

func init() {
	if adminEmail == "" {
		adminEmail = "admin@cially.org"
	}
	if adminPassword == "" {
		adminPassword = "admin123!"
	}
	m.Register(func(app core.App) error {
		// initial settings
		settings := app.Settings()
		settings.Meta.AppName = "Cially"
		settings.Meta.HideControls = true
		settings.Logs.MinLevel = 4
		if err := app.Save(settings); err != nil {
			return err
		}
		// create superuser
		collection, _ := app.FindCollectionByNameOrId(core.CollectionNameSuperusers)
		user := core.NewRecord(collection)
		user.SetEmail(adminEmail)
		user.SetPassword(adminPassword)
		return app.Save(user)
	}, nil)
}
